lib/edtf/parser.y

```diff
@@ -131,6 +131,8 @@ rule
     result = Date.new(val[0]).unspecified!([:day,:month])
   }
 
+  negative_unspecified :  '-' unspecified { result = -val[1] }
```

Comment:
Here `val[1]` is a an array of values, so here you actually invert only `val[1][0]`.